aeron-driver/src/main/c/collections/aeron_bit_set.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2019 Real Logic Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef AERON_AERON_BIT_SET_H
+#define AERON_AERON_BIT_SET_H
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include "util/aeron_bitutil.h"
+
+struct aeron_bit_set_stct
+{
+    size_t bit_count;
+    uint64_t* bits;
+    uint64_t* static_array;
+};
+
+typedef struct aeron_bit_set_stct aeron_bit_set_t;
+
+inline int aeron_bit_set_init(aeron_bit_set_t* bit_set, bool initial_value)
+{
+    const size_t memory_size = ((bit_set->bit_count + 63) / 64) * sizeof(uint64_t);
+    int c = initial_value ? 0xFF : 0;
+    memset(bit_set->bits, c, memory_size);
+    return 0;
+}
+
+inline int aeron_bit_set_stack_alloc(
+    size_t bit_count,
+    uint64_t* static_array,
+    size_t static_array_len,
+    aeron_bit_set_t* bit_set)
+{
+    (bit_set)->bit_count = bit_count;
+    (bit_set)->static_array = static_array;
+
+    const size_t u64_len = ((bit_count + 63) / 64);
```

Comment:
should name be bitset_length?

Reply:
Went with bit_set_length to match the naming of the struct.